src/mapWithFeedback.ts

```diff
@@ -0,0 +1,147 @@
+import { LazyResult, _reduceLazy } from './_reduceLazy';
+import { _toLazyIndexed } from './_toLazyIndexed';
+import { PredIndexedOptional } from './_types';
+import { purry } from './purry';
+
+/**
+ * A callback function receiving the first two parameters of a reduce operation.
+ */
+type Reducer<T, Accumulator> = (
+  accumulator: Accumulator,
+  currentValue: T
+) => Accumulator;
```

Comment:
**note:** we inline the same types in `reduce.ts`, and use similar types in `map.ts`... seems like the letters are `(acc: K, item: T) => K`

Reply:
Thanks! I was looking at some of the more recent PRs and noted they tended to use full type names. I'll stick with the old convention.

Reply:
I usually prefer T for the main "actor" of the type, and full names for all the rest, but regardless we don't have strict conventions on generic variable names, as long as the type is easy to read and "feels" consistent in the codebase